azurecompute/src/test/java/org/jclouds/azurecompute/features/AffinityGroupApiMockTest.java

```diff
@@ -0,0 +1,100 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.features;
+
+import static org.testng.Assert.assertEquals;
+
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+
+import org.jclouds.azurecompute.internal.BaseAzureComputeApiMockTest;
+import org.jclouds.azurecompute.xml.ListAffinityGroupsHandlerTest;
+import org.jclouds.azurecompute.domain.CreateAffinityGroupParams;
+import org.jclouds.azurecompute.domain.UpdateAffinityGroupParams;
+
+import org.testng.annotations.Test;
+
+@Test(groups = "unit", testName = "AffinityGroupApiMockTest")
+public class AffinityGroupApiMockTest extends BaseAzureComputeApiMockTest {
+
+   
+   
+   public void testList() throws Exception {
+      final MockWebServer server = mockAzureManagementServer();
+      server.enqueue(xmlResponse("/affinityGroups.xml"));
+
+      try {
+         final AffinityGroupApi api = api(server.getUrl("/")).getAffinityGroupApi();
+
+         assertEquals(api.list(), ListAffinityGroupsHandlerTest.expected());
+
+         assertSent(server, "GET", "/affinitygroups");
+      } finally {
+         server.shutdown();
+      }
+   }
+
+   public void testAdd() throws Exception {
+      final MockWebServer server = mockAzureManagementServer();
+      server.enqueue(requestIdResponse("request-1"));
+
+      try {
+         final AffinityGroupApi api = api(server.getUrl("/")).getAffinityGroupApi();
+
+         final CreateAffinityGroupParams params = CreateAffinityGroupParams.builder().name("mygroup").label("foo").
+                 location("West Europe").build();
+
+         assertEquals(api.add(params), "request-1");
+
+         assertSent(server, "POST", "/affinitygroups", "/createaffinitygroupparams.xml");
+      } finally {
+         server.shutdown();
+      }
+   }
+
+   public void testUpdate() throws Exception {
+      MockWebServer server = mockAzureManagementServer();
+      server.enqueue(requestIdResponse("request-2"));
+
+      try {
+         final AffinityGroupApi api = api(server.getUrl("/")).getAffinityGroupApi();
+
+         final UpdateAffinityGroupParams params = UpdateAffinityGroupParams.builder().label("foo").
+                 description("mygroup description").build();
+
+         assertEquals(api.update("mygroup", params), "request-2");
+
+         assertSent(server, "PUT", "/affinitygroups/mygroup", "/updateaffinitygroupparams.xml");
+      } finally {
+         server.shutdown();
+      }
+   }
+
+   public void testDelete() throws Exception {
+      MockWebServer server = mockAzureManagementServer();
+      server.enqueue(requestIdResponse("request-3"));
+
+      try {
+         final AffinityGroupApi api = api(server.getUrl("/")).getAffinityGroupApi();
+
+         assertEquals(api.delete("mygroup"), "request-3");
+
+         assertSent(server, "DELETE", "/affinitygroups/mygroup");
+      } finally {
+         server.shutdown();
+      }
+   }
```

Comment:
Add mock tests that exercise the fallbacks you explicitly define in the API class.